public/app/features/dashboard/services/DashboardLoaderSrv.ts

```diff
@@ -110,9 +121,44 @@ abstract class DashboardLoaderSrvBase<T> implements DashboardLoaderSrvLike<T> {
 
     return { data: scriptResult };
   }
+
+  protected loadDashboardFromProvisioning(repo: string, path: string) {
+    const params = new URLSearchParams(window.location.search);
+    const ref = params.get('ref') ?? undefined; // commit hash or branch
+
+    const url = `${BASE_URL}/repositories/${repo}/files/${path}`;
+    return getBackendSrv()
+      .get(url, ref ? { ref } : undefined)
+      .then((v) => {
+        // Load the results from dryRun
+        const dryRun = v.resource.dryRun;
+        if (!dryRun) {
+          return Promise.reject('failed to read provisioned dashboard');
+        }
+
+        if (!dryRun.apiVersion.startsWith('dashboard.grafana.app')) {
+          return Promise.reject('unexpected resource type: ' + dryRun.apiVersion);
+        }
+        if (!this.isVersionSupported(dryRun.apiVersion.split('/')[1])) {
+          throw new DashboardVersionError(dryRun.apiVersion.split('/')[1], 'Unsupported dashboard version');
+        }
+
+        return this.processDashboardFromProvisioning(repo, path, dryRun, {
+          file: url,
+          ref: ref,
+          repo: repo,
+        });
+      });
+  }
 }
 
 export class DashboardLoaderSrv extends DashboardLoaderSrvBase<DashboardDTO> {
+  private SUPPORTED_VERSIONS = ['v1alpha1', 'v0alpha1'];
```

Comment:
why is v2 missing here?

is this something we need to check explicitly?

Reply:
@ryantxu this is loader for handling v1 spec, v2 is handled through a separate implementation DashboardLoaderSrvV2

Reply:
@ryantxu  - removed this, found an easier way.